sympy/printing/c.py

```diff
@@ -396,8 +397,11 @@ def _print_MatrixElement(self, expr):
 
     def _print_Symbol(self, expr):
         name = super()._print_Symbol(expr)
-        # if name is an indexed, e.g. a[0], only check the variable name
-        self._check_valid_c_variable_name(name.split('[')[0])
+        if self._settings['strict']:
```

Comment:
(I could have sworn I wrote a comment, here, github or me messed up, here I go again)
I would prefer if we introduced a new setting for checking the names of the symbols, since the type of checking is somewhat orthogonal to what our current `strict` setting does. What about `strict_symbol_names` or `strict_names`? (The exact naming is not that important to me at least, only that we give users fine grained control).

Reply:
It would also differ with respect to default: we made `strict` default to `True` because it was generating non-functional code before. For this new option, that does not hold, so the default will need to be `False` I think...

Reply:
Yes, I can do that. Although it isn't so clear what 'strict' means in the current code. I couldn't really find an explanation. There is also a special setting for the reserved words check. I'll revisit this when I have some time and make a new proposal.

Reply:
That's my fault. I need write some documentation for what it is doing, and figure out where to put it (if we have a central place where we document printer options for the code printer). Currently it is simply mentioned in passing: https://github.com/sympy/sympy/blob/bb7a8d9316b01c05345d3b69e3f39541df5bf2b6/doc/src/modules/printing.rst?plain=1#L269